### ISSUE TITLE
CSV files with field names that contain spaces

### ISSUE THREAD

**Issue by glynnbird** [ibm-cds-labs/simple-search-service#28]
So imagine we have a data like this
```
firstname,lastname,home address
glynn,bird,10 front street
```

the firstname and lastname fields are ok, but the `home address` field causes us problems.

We try and fudge it in the front end (https://github.com/ibm-cds-labs/simple-search-service/blob/master/public/js/seams.js#L153) to replace spaces with underscores. But the back end needs to do the same because Cloudant doesn't like names of indexes to have spaces in either.

So I tried this change at this line https://github.com/ibm-cds-labs/simple-search-service/blob/master/lib/schema.js#L65:

```js
  for (var i in schema.fields) {
    var f = schema.fields[i];
    var nicename = f.name.replace(/ /g,"_");
    if (f.name != "_id") {
      func += '  indy("' + nicename + '", doc["'+ f.name + '"], ' + f.facet + ');\n'; 
    }     
  }
```

to do the same fudge at the back end. But the this only fudges the index name. 

So then I tried removing the front-end fudge so that it leaves the field name with a space in it, but then I get JS errors at the front-end:

```
jquery.js:1496 Uncaught Error: Syntax error, unrecognized expression: select[name=Business Unit]
```

because selectors with spaces in freak out too.

---------------

In summary

* on the server side, in schema.js, we are going to need to fudge the index name so that it doesn't have spaces in, otherwise Cloudant freaks out
* but we probably want the actual JSON data to retain the field names that were in the file, spaces and all
* and we want to make sure the front end can handle that

**Comment by glynnbird** [comment]
Fixed by pull#29
